src/complex/Filter/Arguments.hpp

```diff
@@ -41,6 +42,13 @@ class COMPLEX_EXPORT Arguments
    */
   void insert(std::string key, std::any&& value);
 
+  /**
+   *
+   * @param key
+   * @param value
+   */
+  void insert(const complex::StringLiteral& key, const std::any& value);
```

Comment:
```suggestion
  void insert(StringLiteral key, const std::any& value);
```
I've been thinking about the best way to go about this. We should probably discuss our options. If we did go this route, `StringLiteral` should be passed by value like `std::string_view`.